src/tap/data/ResultSetTableIterator.java

```diff
@@ -731,7 +731,7 @@ else if (colValue instanceof Boolean)
 			else if (colType != null && colValue != null && colType.type == DBDatatype.CHAR && (colType.length == 1 || colType.length <= 0) && colValue instanceof String)
 				colValue = ((String)colValue).charAt(0);
 			// if the column value is a geometrical object, it must be serialized in STC-S:
-			else if (translator != null && colType != null && colType.isGeometry()){
+			else if ((translator != null && colType != null && colType.isGeometry()) || colValue instanceof byte[]) {
```

Comment:
In this particular case, it was just a incomplete commit: I forgot to adapt the automatic datatype detection for `CENTROID` functions. I have submitted a new commit fixing this.

Can you fetch it and check whether the problem is fixed?